brightray/browser/url_request_context_getter.cc

```diff
@@ -163,6 +162,32 @@ URLRequestContextGetter::~URLRequestContextGetter() {
 #endif
 }
 
+std::unique_ptr<base::CallbackList<void(const CookieDetails*)>::Subscription>
+URLRequestContextGetter::RegisterCookieChangeCallback(
+    const base::Callback<void(const CookieDetails*)>& cb) {
+  return cookie_change_sub_list_.Add(cb);
```

Comment:
@deepak1556 Having the `Subscription` destroyed in UI thread with `CallbackList` destroyed in IO thread could cause race conditions on exit. Using `content::NotificationService` could be a safer choice, like what Chromium did with `chrome::NOTIFICATION_COOKIE_CHANGED_FOR_EXTENSIONS`.

Reply:
I went down that road but soon discovered that `NotificationService` is being removed https://bugs.chromium.org/p/chromium/issues/detail?id=268984 . So chose this instead, moving the ownership of `CallbackList` to `BrowserContext` should fix the thread issue. Thoughts ?

Reply:
> moving the ownership of CallbackList to BrowserContext should fix the thread issue.

This sounds good to me.